server/src/main/java/io/deephaven/server/barrage/BarrageMessageProducer.java

```diff
@@ -361,6 +384,7 @@ public BarrageMessageProducer(final Scheduler scheduler,
         this.updateIntervalMs = updateIntervalMs;
         this.onGetSnapshot = onGetSnapshot;
 
+        this.parentTableSize = parent.size();
```

Comment:
Hmm. Are you sure about this comment? I see the code path calling `getResultNoMemo` which does use `initializeWithSnapshot`... which means it should be safe to ask the parent for its table size.

Reply:
Ah, I see the `snapshotNeeded` method, now. We need some consistent way of knowing the parent table's RowSet and/or size. Size was the cute way to tell if a viewport was satisfied (by comparing to the first key not yet sent).

We do need to record this in case the table does not update. If we miss a tick, then a viewport could be missing a row, too.

Reply:
Yeah, I don't see any alternative; we need to get rid of the `snapshotNeeded()` override. The unfortunate part being that this was the only one. :-(